### PR TITLE
Clarify recommended use of metadata in docs.

### PR DESCRIPTION
**Pull request by craigcitro**
(This is a followup to #274.)

/cc @rgbkrk

### CODE REVIEW

File: docs/api/kernelspec.rst [hunk 1]
```diff
@@ -30,6 +30,10 @@ kernelspec - discovering kernels
       Additional kernel-specific metadata; clients can use this as needed,
       for instance to aid in kernel selection and filtering.
 
+      As with all other metadata fields, jupyter clients should use
+      app-specific submessages as opposed to storing their own top-level
```
Comment (takluyver) [inline docs/api/kernelspec.rst]: submessages -> subdictionaries or subobjects?

File: docs/api/kernelspec.rst [hunk 2]
```diff
@@ -30,6 +30,10 @@ kernelspec - discovering kernels
       Additional kernel-specific metadata; clients can use this as needed,
       for instance to aid in kernel selection and filtering.
 
+      As with all other metadata fields, jupyter clients should use
```
Comment (takluyver) [inline docs/api/kernelspec.rst]: Do we specifically mean 'clients' here, or just third party tools in general?

Comment (rgbkrk) [inline docs/api/kernelspec.rst]: Clients as in any _jupyter protocol_ frontend, not necessarily a jupyter _branded_ client.

File: docs/api/kernelspec.rst (hunk 1 above)
Comment (rgbkrk) [inline docs/api/kernelspec.rst]: What does this mean?

Comment (takluyver) [inline docs/api/kernelspec.rst]: The kernelspec is not a message, so I wouldn't call parts of its data 'submessages'.

File: docs/api/kernelspec.rst (hunk 2 above)
Comment (craigcitro) [inline docs/api/kernelspec.rst]: Improved (I hope). LMK if this is clearer.

File: docs/api/kernelspec.rst (hunk 1 above)
Comment (craigcitro) [inline docs/api/kernelspec.rst]: Add an example, in the hopes that would clarify things. (Happy to tweak more if it doesn't.)

**Comment by craigcitro** [comment]
Sorry for the slow reply; I think the filters I set up for my github emails are working a little **too** well. 😊

**Comment by minrk** [comment]
Thanks!

**Pull request merged**